core/p2p/base/option.go

```diff
@@ -7,12 +7,19 @@ type MsgOptions struct {
 	TargetPeerIDs   []string
 	Percentage      float32 //percentage wait for return
 	// compress
-	Compress bool
+	Compress  bool
+	WhiteList map[string]bool
 }
 
 // MessageOption define single option function
 type MessageOption func(*MsgOptions)
 
+func WithWhiteList(whiteList map[string]bool) MessageOption {
```

Comment:
Why not use WithTargetPeerAddrs or WithTargetPeerIDs?

Reply:
Different options, different effects.